htdocs/core/ajax/ajaxtooltip.php

```diff
@@ -0,0 +1,200 @@
+	$langs->load('supplier_proposal');
+	$classpath = 'supplier_proposal/class';
+} elseif ($objecttype == 'shipping') {
+	$langs->load('sendings');
+	$classpath = 'expedition/class';
+	$myobject = 'expedition';
+	$module = 'expedition_bon';
+} elseif ($objecttype == 'delivery') {
+	$langs->load('deliveries');
+	$classpath = 'delivery/class';
+	$myobject = 'delivery';
+	$module = 'delivery_note';
+} elseif ($objecttype == 'contract') {
+	$langs->load('contracts');
+	$classpath = 'contrat/class';
+	$module = 'contrat';
+	$myobject = 'contrat';
+} elseif ($objecttype == 'member') {
+	$langs->load('members');
+	$classpath = 'adherents/class';
+	$module = 'adherent';
+	$myobject = 'adherent';
+} elseif ($objecttype == 'cabinetmed_cons') {
+	$classpath = 'cabinetmed/class';
+	$module = 'cabinetmed';
+	$myobject = 'cabinetmedcons';
+} elseif ($objecttype == 'fichinter') {
+	$langs->load('interventions');
+	$classpath = 'fichinter/class';
+	$module = 'ficheinter';
+	$myobject = 'fichinter';
+} elseif ($objecttype == 'project') {
+	$langs->load('projects');
+	$classpath = 'projet/class';
+	$module = 'projet';
+} elseif ($objecttype == 'task') {
+	$langs->load('projects');
+	$classpath = 'projet/class';
+	$module = 'projet';
+	$myobject = 'task';
+} elseif ($objecttype == 'stock') {
+	$classpath = 'product/stock/class';
+	$module = 'stock';
+	$myobject = 'stock';
+} elseif ($objecttype == 'inventory') {
+	$classpath = 'product/inventory/class';
+	$module = 'stock';
+	$myobject = 'inventory';
+} elseif ($objecttype == 'mo') {
+	$classpath = 'mrp/class';
+	$module = 'mrp';
+	$myobject = 'mo';
+} elseif ($objecttype == 'productlot') {
+	$classpath = 'product/stock/class';
+	$module = 'stock';
+	$myobject = 'productlot';
+}
+
+// Generic case for $classfile and $classname
+$classfile = strtolower($myobject);
+$classname = ucfirst($myobject);
+//print "objecttype=".$objecttype." module=".$module." subelement=".$subelement." classfile=".$classfile." classname=".$classname." classpath=".$classpath;
+
+if ($objecttype == 'invoice_supplier') {
+	$classfile = 'fournisseur.facture';
+	$classname = 'FactureFournisseur';
+	$classpath = 'fourn/class';
+	$module = 'fournisseur';
+} elseif ($objecttype == 'order_supplier') {
+	$classfile = 'fournisseur.commande';
+	$classname = 'CommandeFournisseur';
+	$classpath = 'fourn/class';
+	$module = 'fournisseur';
+} elseif ($objecttype == 'supplier_proposal') {
+	$classfile = 'supplier_proposal';
+	$classname = 'SupplierProposal';
+	$classpath = 'supplier_proposal/class';
+	$module = 'supplier_proposal';
+} elseif ($objecttype == 'stock') {
+	$classpath = 'product/stock/class';
+	$classfile = 'entrepot';
+	$classname = 'Entrepot';
+} elseif ($objecttype == 'facturerec') {
+	$classpath = 'compta/facture/class';
+	$classfile = 'facture-rec';
+	$classname = 'FactureRec';
+	$module = 'facture';
+} elseif ($objecttype == 'mailing') {
+	$classpath = 'comm/mailing/class';
+	$classfile = 'mailing';
+	$classname = 'Mailing';
+}
+
```

Comment:
@frederic34 @eldy 
Considering the constant needs in the core of dolibarr or in the external modules of a correspondence of the names of element or path when a module goes out of "standards", shouldn't we inform one or more constants when activating a module that would give us all this information instead of having to do some hard code to reference the exceptions?

Reply:
objects with classfortooltip will use the existing tooltip, the new tooltip use classforajaxtooltip

Reply:
@frederic34 je vais te la faire en Français ;-)

Vu les besoins constants dans le cœur de dolibarr ou dans les modules externes d'une correspondance des noms d'élément ou de chemin lorsqu'un module sort des "normes", ne faut-il pas renseigner une ou plusieurs constantes lors de l'activation d'un module qui donnerait nous toutes ces informations au lieu d'avoir à faire du code dur pour référencer les exceptions ?

genre : propale ->propal , projet -> project
et autres répertoires qui ne correspondent pas au nom de l'élément

Reply:
If the module is non-standard, then it must adapt, otherwise it keeps its tooltips unchanged

Reply:
@frederic34 je ne parle pas de ton code, je parle de ceci ci-dessous en général : 

```
elseif ($objecttype == 'shipping') {
	$langs->load('sendings');
	$classpath = 'expedition/class';
	$myobject = 'expedition';
	$module = 'expedition_bon';
```

ces correspondances "nom element", "chemin module", "nom de la classe", etc... qu'on retrouve dans pas mal d'endroit en dur dans le code et aussi dans les modules externes pour être compatible quand le module n'a pas les "normes"... Donc je me disait qu'on pourrait instaurer une constante du type json qui pourrait emmagasiner toutes ces infos lors de l'activation d'un module, histoire de ne pas répéter les mêmes correspondances un peu partout ?